### PR TITLE
etcdserver: reject member removal that breaks active quorum

### PR DESCRIPTION
**Pull request by heyitsanthony**

### CODE REVIEW

File: etcdserver/server.go [hunk 1]
```diff
@@ -857,6 +856,31 @@ func (s *EtcdServer) RemoveMember(ctx context.Context, id uint64) error {
 	return s.configure(ctx, cc)
 }
 
+func (s *EtcdServer) mayRemoveMember(id uint64) error {
+	if !s.Cfg.StrictReconfigCheck {
+		return nil
+	}
+
+	if !s.cluster.IsReadyToRemoveMember(id) {
+		plog.Warningf("not enough started members, rejecting remove member %v", id)
```
Comment (gyuho) [inline etcdserver/server.go]: `%s", types.ID(id))`, so that the log output format can be consistent with other hex ids?

**Comment by xiang90** [comment]
awesome! lgtm.

**Comment by gyuho** [comment]
LGTM. Thanks!

File: etcdserver/server.go (hunk 1 above)
Comment (heyitsanthony) [inline etcdserver/server.go]: sure

**Comment by xiang90** [comment]
Can we add release note label to this? This is a user facing change

**Comment by gyuho** [comment]
@xiang90 Just added. Will make sure we highlight this.

**Pull request merged**